bin/cylc-show-times

```diff
@@ -0,0 +1,372 @@
+            df_describe.index.rename(None, inplace=True)
+            for timing_category in self.df.columns:
+                self.write_category(
+                    buf, timing_category, df_reshape, df_describe
+                )
+        self.write_summary_footer(buf)
+
+    def write_summary_header(self, buf):
+        pass
+
+    def write_summary_footer(self, buf):
+        pass
+
+    def write_group_header(self, buf, group):
+        pass
+
+    def write_category(self, buf, category, df_reshape, df_describe):
+        pass
+
+    def _check_imports(self):
+        try:
+            import pandas as pd
+        except ImportError:
+            msg = 'Cannot import pandas - summary unavailable.'
+            sys.stderr.write(msg)
+            raise Exception(msg)
+
+    @staticmethod
+    def _dt_to_s(dt):
+        import pandas as pd
+        try:
+            return dt.total_seconds()
+        except AttributeError:
+            # Older versions of pandas have the timedelta as a numpy
+            # timedelta64 type, which didn't support total_seconds
+            return pd.Timedelta(dt).total_seconds()
+
+
+class TextTimingSummary(TimingSummary):
+    """Timing summary in text form."""
+
+    line_width = 80
+
+    def write_group_header(self, buf, group):
+        title = 'Host: %s\tBatch System: %s' % group
+        buf.write('=' * self.line_width + '\n')
+        buf.write(title.center(self.line_width-1) + '\n')
+        buf.write('=' * self.line_width + '\n')
+
+    def write_category(self, buf, category, df_reshape, df_describe):
+        buf.write(category.center(self.line_width) + '\n')
+        buf.write(('-'*len(category)).center(self.line_width) + '\n')
+        buf.write(df_describe[category].to_string())
+        buf.write('\n\n')
+
+
+class HTMLTimingSummary(TimingSummary):
+    """Timing summary in HTML form."""
+
+    def write_summary_header(self, buf):
+        css = """
+            body {
+                font-family: Sans-Serif;
+                text-align: center;
+            }
+            h1 {
+                background-color: grey;
+            }
+            h2 {
+                width: 85%;
+                background-color: #f0f0f0;
+                margin: auto;
+            }
+            table {
+                width: 75%;
+                margin: auto;
+                border-collapse: collapse;
+            }
+            tr:nth-child(even) {
+                background-color: #f2f2f2;
+            }
+            td {
+                text-align: right;
+            }
+            th, td {
+                border-bottom: 1px solid grey;
+            }
+            svg {
+                width: 65%;
+                height: auto;
+                margin: auto;
+                display: block;
+            }
+            .timing {
+                padding-bottom: 40px;
+            }
+        """
+
+        buf.write('<html><head><style>%s</style></head><body>' % css)
```

Comment:
Just wondering if you'd consider moving the display presentation logic into Jinja2 templates?